internal/infrastructure/kubernetes/deployment.go

```diff
@@ -27,13 +29,102 @@ const (
 	// envoyCfgVolMntDir is the directory name of the Envoy configuration volume.
 	envoyCfgVolMntDir = "config"
 	// envoyCfgFileName is the name of the Envoy configuration file.
-	envoyCfgFileName = "envoy.json"
+	envoyCfgFileName = "bootstrap.yaml"
 	// envoyHTTPPort is the container port number of Envoy's HTTP endpoint.
 	envoyHTTPPort = int32(8080)
 	// envoyHTTPSPort is the container port number of Envoy's HTTPS endpoint.
 	envoyHTTPSPort = int32(8443)
 )
 
+var bootstrapTmpl = template.Must(template.New(envoyCfgFileName).Parse(`
```

Comment:
I'd consider putting this in a separate `bootstrap.yaml` file and saying 

```go
//go:embed bootstrap.yaml
const string bootstrapTmplStr

var bootstrapTmpl = template.Must(template.New(envoyCfgFileName).Parse(bootstrapTmplStr))
```
so that the file is easier to edit, with editors recognizing it as yaml.